src/N3Lexer.js

```diff
@@ -300,12 +300,33 @@ export default class N3Lexer {
       case ']':
       case '(':
       case ')':
-      case '{':
       case '}':
         if (!this._lineMode) {
           matchLength = 1;
           type = firstChar;
+          break;
         }
+      case '{':
+        if (!this._lineMode) {
+          // We need at least 2 tokens lookahead to distinguish "{|" and "{ "
+          if (input.length < 2)
+            break;
+
+          // Try to find a quoted triple annotation start
+          if (input[1] === '|') {
+            type = '{|', matchLength = 2;
+            break;
+          }
```

Comment:
Having this inside the `!this._lineMode` condition implies that annotation syntax is not supported in the n-quads flavour of RDF-star.

Is this correct?

Note this `{` case is currently implemented as follows in the staged v2 branch

```ts
        // We can properly evaluate this case if we are
        // not in rdfStar mode or we can look ahead to
        // see if there is a pipe following the {
        const long = input.length !== 1;
        if (long && input[1] === '|')
          matchLength = 2, type = '{|';
        else if (!this._lineMode && (long || !this._supportsRDFStar))
          matchLength = 1, type = firstChar;
        break;
```

Reply:
AFAICS, N-Triples and N-Quads do not support annotated triples: https://w3c.github.io/rdf-star/cg-spec/editors_draft.html#n-triples-star

This makes sense, given that annotated triples are just syntactic sugar, and N-Triples and N-Quads have not provided the previously-existing syntactical shortcuts that Turtle and TriG had so far.

Reply:
Makes sense; and the [n-quads test suite](https://w3c.github.io/rdf-star/tests/nt/syntax/) doesn't test for it.

I've opened up https://github.com/w3c/rdf-star/issues/279 requesting some negative tests for this as well.